middleware/packet-forward-middleware/router/keeper/keeper.go

```diff
@@ -370,6 +370,7 @@ func (k *Keeper) RetryTimeout(
 	}
 
 	if data.Memo != "" {
+		metadata.Next = &types.JSONObject{}
 		if err := json.Unmarshal([]byte(data.Memo), metadata.Next); err != nil {
 			return fmt.Errorf("error unmarshaling memo json: %w", err)
```

Comment:
I'm taking a second look at this in the context of https://github.com/osmosis-labs/osmosis/issues/5969. It seems like some packets are returning this error. I think that error should never happen: if the unmarshaling fails we should just return a json object that is a string.

This should in theory be what's happening inside `JSONObject.UnmarshalJSON()`, but I think it needs to be reviewed as it only accepts json objects or true/false and not arbitrary strings